nixos/modules/services/networking/i2pd.nix

```diff
@@ -222,14 +222,12 @@ let
         in concatStringsSep "\n" inTunOpts))];
     in pkgs.writeText "i2pd-tunnels.conf" opts;
 
-  i2pdSh = pkgs.writeScriptBin "i2pd" ''
-    #!/bin/sh
-    exec ${cfg.package}/bin/i2pd \
-      ${if cfg.address == null then "" else "--host="+cfg.address} \
-      --service \
-      --conf=${i2pdConf} \
-      --tunconf=${tunnelConf}
-  '';
+  i2pdFlags = toString [
+    (optionalString (cfg.address != null) ("--host=" + cfg.address))
```

Comment:
Since we already have a list, it would be better to use `[...] ++ lib.optionals <condition> [...]` to avoid an empty element in the list.

Reply:
Good call.